doc/tutorials/gpu-passthru.rst

```diff
@@ -0,0 +1,147 @@
+Intel GVT-d is one flavor of graphics virtualization approaches also
+names as Intel-Graphics-Device pass-through feature.
+It based on Intel VT-d technology,
+and some special graphics related configuration.
+This flavor allows direct assignment of an entire GPU’s prowess to a single user,
+passing the native driver capabilities through the hypervisor without any limitations.
+
+Hardware Platform Supported
+***************************
+Currently, ACRN has enabled GVT-d on the following platforms:
+
+* Kaby Lake
+* Whiskey Lake
+* Elkhart Lake
+
+BIOS Setting
+**********************
+
+Kaby Lake Platform
+==================
+* set "IGD Minimum Memory" to "64MB" in "Devices → Video → IGD Minimum Memory".
+
+Whiskey Lake Platform
+=====================
+* set "PM Support"  to "Enabled" in "chipset → System Agent (SA) Configuration → Graphics Configuration → PM support".
+* set "DVMT Pre-Allocated" to "64MB" in "chipset → System Agent (SA) Configuration → Graphics Configuration → DVMT Pre-Allocated"
+
+Elkhart Lake Platform
+=====================
+* set "DMVT Pre-Allocated" to "64MB" in "Intel Advanced Menu → System Agent(SA) Configuration → Graphics Configuration → DMVT Pre-Allocated"
+
+Pass-through GPU to Guest
+**************************
+
+1. Copy ``/usr/share/acrn/samples/nuc/launch_win.sh`` to ``install_win.sh``
+
+::
+
+   cp /usr/share/acrn/samples/nuc/launch_win.sh ~/install_win.sh
+
+2. Modify ``install_win.sh`` script to specify the Windows image you use.
+
+3. Modify ``install_win.sh`` script to enable GVT-d.
+
+Add the following commands before ``acrn-dm -A -m $mem_size -s 0:0,hostbridge \``
+
+::
+
+      gpudevice=`cat /sys/bus/pci/devices/0000:00:02.0/device`
+
+      echo "8086 $gpudevice" > /sys/bus/pci/drivers/pci-stub/new_id
+      echo "0000:00:02.0" > /sys/bus/pci/devices/0000:00:02.0/driver/unbind
+      echo "0000:00:02.0" > /sys/bus/pci/drivers/pci-stub/bind
+
+Replace ``-s 2,pci-gvt -G "$2" \`` with ``-s 2,passthru,0/2/0,gpu \``
+
+4. Run the ``launch_win.sh``.
+
+      .. note:: If you want to pass-through GPU to Clear Linux User VM, the steps are
+         the same as above except your script.
+
+Enable GVT-d GOP driver
+***********************
+
+Issue
+======
+When enabling GVT-d, guest OS couldn’t light up physical screen
+before OS driver load. So guest BIOS and
+grub GUI couldn’t be showed on the physical screen.
+
+Reason
+==========
+Physical display is initialized by GOP driver or VBIOS
+before OS driver load, but guest BIOS doesn’t have them.
+
+Solution
+==========
+Integrate GOP driver binary into OVMF as a Dxe driver,
+then guest OVMF could see GOP driver and run it in
+graphic pass-through environment.
+So physical display could be initialized
+by GOP and used by guest BIOS and guest grub.
+
+Steps
+======
+1. fetch ACRN OVMF
+
+::
+
+      git clone https://github.com/projectacrn/acrn-edk2.git
+
+2. fetch vbt and gop driver
+
+Fetch vbt and gop driver from the vendor according to your CPU model name.
```

Comment:
Can we provide some more detailed instructions here? 

Since we are talking about GVT-d in this doc, I assume this only applies to Intel GPU, and so perhaps we can give a URL where the user can download these from?

Reply:
These binaries aren't public. So we can't give the link.

Reply:
How can one get those? Is there an Intel support website where one can register and download those?

Reply:
We can get those from vendor only.

Reply:
The vendor means: vendor that flashes BIOS into the board.

Reply:
OK. Where do I get these files for my NUC7i5BNH system for example?

The couple of patches you ask to apply are also linked to these VBT and GOP drivers. Are these names always going to be consistent no matter the system vendor?

Reply:
> OK. Where do I get these files for my NUC7i5BNH system for example?
> 
I get the vendor information from PM. I will discuss with this description here with my colleagues.
> The couple of patches you ask to apply are also linked to these VBT and GOP drivers. Are these names always going to be consistent no matter the system vendor?
These file names are determined by the vendor. The customer can check with the vendor.

Reply:
> OK. Where do I get these files for my NUC7i5BNH system for example?
> 
I get the vendor information from PM. I will discuss with this description here with my colleagues.
> The couple of patches you ask to apply are also linked to these VBT and GOP drivers. Are these names always going to be consistent no matter the system vendor?
>

The file names are determined by the vendor. The customer can confirm this with the vendor.

Reply:
Hi, @gvancuts The customer should know the board vendor information. It isn't our business here.

Reply:
I understand that part. I would like to try out the steps in this tutorial, and run that on my NUC7i5BNH system. Who can I contact to get these files?

Reply:
Please add a note in the tutorial that your patches will have to be adjusted if the files provided by the vendor have a different filename.

Reply:
> Hi, @gvancuts The customer should know the board vendor information. It isn't our business here.

Is it the board manufacturer or the bios vendor that should be contacted?

Reply:
> I understand that part. I would like to try out the steps in this tutorial, and run that on my NUC7i5BNH system. Who can I contact to get these files?

You can connect Nanlin for help

Reply:
> Please add a note in the tutorial that your patches will have to be adjusted if the files provided by the vendor have a different filename.

Ok. I will add the note

Reply:
> > Hi, @gvancuts The customer should know the board vendor information. It isn't our business here.
> 
> Is it the board manufacturer or the bios vendor that should be contacted?

board manufacturer

Reply:
> board manufacturer

Thanks for the clarification. Can you update the text to make this clearer?

Reply:
> > board manufacturer
> 
> Thanks for the clarification. Can you update the text to make this clearer?

Sure. I will update this PR.